source/documentation/deploying-an-app/github-actions-continuous-deployment.html.md.erb

```diff
@@ -0,0 +1,430 @@
+
+### Push to ECR
+
+The GitHub Action needs to push to your ECR, so it's going to need the AWS
+credentials.
+
+Add these to your GitHub repository as [GitHub Secrets] by going to the
+"Settings" page for your repository and clicking "Secrets" in the left-hand
+navigation.
+
+Create new secrets with the following names, and the values for your ECR:
+
+* AWS_ACCESS_KEY_ID
+* AWS_SECRET_ACCESS_KEY
+
+Now add this step to the end of your GitHub Action:
+
+```yaml
+      - name: Push to ECR
+        id: ecr
+        uses: jwalton/gh-ecr-push@v1
+        with:
+          access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          region: eu-west-2
+          local-image: foo
+          image: ${ECR}:${{ github.sha }}
+
+```
+
+That takes care of building the docker image and pushing it to the ECR - in
+this case, tagged with the SHA hash of the last commit to the GitHub
+repository.
+
+### Update the manifest
+
+We need to update the kubernetes manifest so that it deploys the appropriate tag version
+of the docker image.
+
+We can do this by turning our `kubernetes-deploy.yaml` file into a template,
+and interpolating the image tag at deployment time.
+
+We'll use
+[envsubst](https://manpages.ubuntu.com/manpages/trusty/man1/envsubst.1.html)
+for this. Any other templating system will work, but envsubst is easiest to set
+up for this tutorial, because it's already installed on the GitHub actions
+`ubuntu-latest` VM.
+
+* Rename the file from `kubernetes-deploy.yaml` to `kubernetes-deploy.tpl`
+
+* Change the `image:` line to this:
+
+```yaml
+image: 754256621582.dkr.ecr.eu-west-2.amazonaws.com/${ECR}:${IMAGE_TAG}
+```
+
+The `ECR` environment variable exists already, so we just need to supply
+`IMAGE_TAG`.
+
+* Add this step to the end of the GitHub Action:
+
+```yaml
+      - name: Update image tag
+        run: export IMAGE_TAG=${{ github.sha }} && cat kubernetes-deploy.tpl | envsubst > kubernetes-deploy.yaml
+```
+
+At this point, the updated docker image has been built and pushed to the ECR,
+and the kubernetes manifest has been update with the correct image tag. Now we
+need to apply the updated manifest to the kubernetes cluster.
+
+### Creating a ServiceAccount
+
+You applied the manifest to the cluster using `kubectl` earlier. This worked
+because you are already authenticated to the cluster, but the GitHub Action
+isn't, so we need to provide some credentials and get it to authenticate.
+
+We'll use a [ServiceAccount] for this.
+
+To create a serviceaccount for your namespace, you need to raise a PR against
+the [environments repository]. You can do this with the [cloud platform cli].
+
+Change directories to your namespace folder in a checkout of the environments
+repository, e.g.
+
+```
+namespaces/live-1.cloud-platform.service.justice.gov.uk/dstest
+```
+
+...and run this:
+
+```bash
+cloud-platform environment serviceaccount create
```

Comment:
As this isn't a standard tool, I think we should link to the install sub-heading on the README. If I was coming into this blind, I would find the mention of this tool confusing.

Reply:
There is a link to [this page](https://user-guide.cloud-platform.service.justice.gov.uk/documentation/getting-started/cloud-platform-cli.html#the-cloud-platform-cli) just above the code snippets.

![Screenshot 2021-01-06 at 13 15 44](https://user-images.githubusercontent.com/2338/103728438-5e3b0080-5021-11eb-976c-7a0ff0640912.png)

Do you think that's not enough? I don't think I can make the text **inside** the code snippet into a link, unfortunately.

Reply:
Okay, that should be fine.